src/main/java/io/kamax/matrix/client/AMatrixHttpClient.java

```diff
@@ -269,4 +285,31 @@ protected URI getMediaPath(String action) {
     protected HttpEntity getJsonEntity(Object o) {
         return EntityBuilder.create().setText(gson.toJson(o)).setContentType(ContentType.APPLICATION_JSON).build();
     }
+
+    private void login() {
+        HttpPost request = new HttpPost(getLoginClientPath());
+        _MatrixID user = context.getUser();
```

Comment:
Login doesn't require a Matrix ID, only a localpart. This is a signed of flowed implementation.